unix/syscall_unix.go

```diff
@@ -101,6 +101,7 @@ type mmapper struct {
 	active map[*byte][]byte // active mappings; key is last byte in mapping
 	mmap   func(addr, length uintptr, prot, flags, fd int, offset int64) (uintptr, error)
 	munmap func(addr uintptr, length uintptr) error
+	mremap func(oldaddr uintptr, oldlength uintptr, newlength uintptr, flags int, newaddr uintptr) (xaddr uintptr, err error)
```

Comment:
On non-Linux systems this will be nil and Mremap is going to panic when it tries to call a nil function pointer. Is that the desired behavior?

Alternative ideas:
1) Return an error in Mremap if it's not supported, ie. if mremap == nil.
2) Simulate an Mremap for non-Linux systems that calls munmap followed by mmap.

1 is probably best.

Reply:
Maybe I'm wrong, but this function is part of the mapper, which is private for this package. We are later using this method in exported method in syscall_linux.go. So users won't be able to call it, generally speaking. Am I wrong?

Reply:
Oh, my mistake. I missed that.